pkg/synopsysctl/cmd_update.go

```diff
@@ -76,34 +77,80 @@ var updateOperatorCmd = &cobra.Command{
 		if len(namespace) > 0 {
 			updateOperator(namespace, cmd)
 		} else {
-			namespace := DefaultOperatorNamespace
 			var err error
+			operatorNamespace := DefaultOperatorNamespace
 			if !cmd.LocalFlags().Lookup("mock").Changed && !cmd.LocalFlags().Lookup("mock-kube").Changed {
 				isClusterScoped := util.GetClusterScope(apiExtensionClient)
 				if isClusterScoped {
 					namespace, err = util.GetOperatorNamespace(kubeClient, metav1.NamespaceAll)
 					if err != nil {
 						log.Error(err)
-						return nil
+					}
+					if metav1.NamespaceAll != namespace {
+						operatorNamespace = namespace
 					}
 				}
 			}
-			updateOperator(namespace, cmd)
+			updateOperator(operatorNamespace, cmd)
 		}
 		return nil
 	},
 }
 
 func updateOperator(namespace string, cmd *cobra.Command) error {
 	log.Infof("updating the Synopsys Operator instance in namespace '%s'...", namespace)
-	// Create new Synopsys-Operator SpecConfig
+	var isClusterScoped bool
+	crds := []string{}
+	crdMap := make(map[string]string)
+	// check whether the Synopsys Operator namespace exist
+	if !cmd.LocalFlags().Lookup("mock").Changed && !cmd.LocalFlags().Lookup("mock-kube").Changed {
+		cm, err := util.GetConfigMap(kubeClient, namespace, "synopsys-operator")
+		if err != nil {
+			log.Errorf("unable to find the 'synopsy-operator' config map in namespace '%s' due to %+v", namespace, err)
+			return nil
+		}
+		data := cm.Data["config.json"]
+		var testMap map[string]interface{}
+		err = json.Unmarshal([]byte(data), &testMap)
+		if err != nil {
+			return fmt.Errorf("unable to unmarshall config map data due to %+v", err)
+		}
+		if _, ok := testMap["IsClusterScoped"]; ok {
+			configData := &protoform.Config{}
+			err = json.Unmarshal([]byte(data), &configData)
+			if err != nil {
+				return fmt.Errorf("unable to unmarshall config map data due to %+v", err)
+			}
+			isClusterScoped = configData.IsClusterScoped
+			crds = strings.Split(configData.CrdNames, ",")
+			for _, crd := range crds {
+				crdMap[crd] = crd
```

Comment:
Shouldn't this value be the scope of the crd?

Reply:
@rrati  Instead of going through the array to check the existence, creating the map. Name and Name should be fine here

Reply:
Comment says "// list the existing CRD's and convert them to map with key as name and scope as value" for creation in the else block.  Name:Name is fine if we don't care about the version, just update comments to reflect that to remove confusion

Reply:
:-)

Reply:
done